packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -105,6 +105,12 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
       attachProps(node, this.props, prevProps);
     }
 
+    componentWillUnmount() {
+      if (this.ref.current) {
+        this.ref.current.dismiss();
```

Comment:
I'm getting a console error about changing state on an unmounted component when using the Replace or Push operations. I'm guessing `dismiss` is trying to update things after the modal has already been unmounted; maybe it needs to be `await`ed? 🤔 

Test repo/branch I'm using (need to install the dev build): https://github.com/amandaejohnston/sandbox-ionic-react/tree/FW-1889
Screencast: https://user-images.githubusercontent.com/90629384/200869178-3e9a8130-c1b9-42be-84b7-d870bd518887.mp4

Reply:
`componentWillUnmount` is synchronous, so `await` won't do anything here. We might need to add logic here to only update the state if the component is not actively unmounting: https://github.com/ionic-team/ionic-framework/blob/1ea0cd29d93eb29561c016c215e79f182e07ce8b/packages/react/src/components/createInlineOverlayComponent.tsx#L66-L99 (I'm guessing this is where the state update is coming from, but it could be somewhere else too)

Reply:
This issue occurs because the `onDidDismiss` fires and the `setState` in the app implementation updates after the component unmounts.

This resolves it in implementations: https://github.com/ionic-team/ionic-framework/pull/26245/files#diff-6c9571ce83a153612a7dcb0a9ced4fb6eaf6f3a613c9e7d4c614136311be6173R14

For a Framework fix, we could look at not handling `onDidDismiss` if the component has already unmounted.

Reply:
If it's possible to handle this in the framework, I would rather go that route, since otherwise this could be considered a breaking change 🤔